File: Data Structures & Algorithms/last-stone-weight/submission-1.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq;
        for(auto i: stones) {
            pq.push(i);
        }

        while(pq.size()>1) {
            int f = pq.top();
            pq.pop();
            int s = pq.top();
            pq.pop();
            if(f==s) continue;
            pq.push(abs(f-s));
        }

        return pq.empty() ? 0: pq.top();
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-1.java
class Solution {
    private double pow(double x, int n) {
        if(n==0) return 1;
        double val = myPow(x, n/2);
        val*= val;
        if(n%2==1) val*=x;
        return val;
    }
    public double myPow(double x, int n) {
        int ab = Math.abs(n);
        double pw = 1;
        if(n>0) pw *= pow(x, ab);
        else pw /= pow(x, ab);
        return pw;
    }
}